python/ray/serve/deployment_state.py

```diff
@@ -1330,12 +1331,11 @@ def _check_curr_status(self) -> bool:
             )
             == 0
         ):
-            # Check for deleting.
-            if target_replica_count == 0 and all_running_replica_cnt == 0:
+            if self._deployment_to_delete:
                 return True
 
             # Check for a non-zero number of deployments.
-            elif target_replica_count == running_at_target_version_replica_cnt:
+            if target_replica_count == running_at_target_version_replica_cnt:
```

Comment:
Why change from `elif` to `if` here?

Reply:
ha, I don't see why we need to use elif, since the first if is `return` and there is no relation between the two if now.

Reply:
Ah, good point, the `if` makes sense.